src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java

```diff
@@ -58,6 +61,200 @@ public static void exportAllHistoricalToFile(
         stringHistorical);
   }
 
+  /**
+   * Prepare Read Historical Log Export Bloc Descriptor (EBD) string.
+   *
+   * @param startTime start time of export
+   * @param endTime end time of export
+   * @param includeTagGroupA include tag group A
+   * @param includeTagGroupB include tag group B
+   * @param includeTagGroupC include tag group C
+   * @param includeTagGroupD include tag group D
+   * @param stringHistorical export string historical logs if true
+   * @return EBD string
+   */
+  private static String prepareHistoricalFifoReadEBDString(
+      String startTime,
+      String endTime,
+      boolean includeTagGroupA,
+      boolean includeTagGroupB,
+      boolean includeTagGroupC,
+      boolean includeTagGroupD,
+      boolean stringHistorical) {
+
+    // Check for valid group selection
+    if (!includeTagGroupA && !includeTagGroupB && !includeTagGroupC && !includeTagGroupD) {
+      throw new IllegalArgumentException(
+          "Cannot generate historical logs with no tag groups selected.");
+    }
+
+    // Build string of tag groups for filter type
+    String tagGroupFilterStr = "";
+    if (includeTagGroupA) {
+      tagGroupFilterStr += "A";
+    }
+    if (includeTagGroupB) {
+      tagGroupFilterStr += "B";
+    }
+    if (includeTagGroupC) {
+      tagGroupFilterStr += "C";
+    }
+    if (includeTagGroupD) {
+      tagGroupFilterStr += "D";
+    }
+
+    // Get EBD data type
+    String ebdDataType;
+    if (stringHistorical) {
+      ebdDataType = "HS";
+    } else {
+      ebdDataType = "HL";
+    }
+
+    /*
+     * Build EBD string with parameters
+     * dt[ebdDataType]: data type, value specified in string ebdDataType
+     * ftT: file type, text
+     * startTime: start time for data
+     * endTime: end time for data
+     * flABCD: filter type, specified tag groups
+     */
+    return "$dt"
+        + ebdDataType
+        + "$ftT$st"
+        + startTime
+        + "$et"
+        + endTime
+        + "$fl"
+        + tagGroupFilterStr;
+  }
+
+  /**
+   * Parses Export Block Descriptor Historical Log response into dataPoints. Note: this function
+   * only handles Historical List responses
+   *
+   * @param exporter EBD Exporter
+   * @return dataPoints
+   * @throws IOException for parsing Exceptions
+   * @throws JSONException for JSON parsing Exceptions
+   * @throws TimeoutException timeout for reading event file
+   * @throws CircularizedFileException when event log indicates circularized event
+   */
+  private static ArrayList parseEBDHistoricalLogExportResponse(Exporter exporter)
+      throws IOException, JSONException, TimeoutException, CircularizedFileException {
+    final DataInputStream dataStream = new DataInputStream(exporter);
```

Comment:
I think this should also follow the simpler design offered in my comment on the event file manager https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/pull/16#discussion_r798914601

Reply:
changed.

Reply:
I left a comment about this someplace else, but my main concern is memory efficiency. Normally simpler is more efficient, but I know that StringUtils makes additional copies. I hope that buffered reader is more efficient.

Reply:
It may be the \n character or some other control character in the ASCII table.

Reply:
I replied to your other comment about memory efficiency/performance and included a link that references performance:

--------------------------------

Performance tests

Performance tests for small String (length = 175), url in [github](https://github.com/Vedenin/useful-java-links/blob/master/helloworlds/5.0-other-examples/src/main/java/other_examples/ConvertInputStreamToStringBenchmark.java) (mode = Average Time, system = Linux, score 1,343 is the best):

```
              Benchmark                         Mode  Cnt   Score   Error  Units
 8. ByteArrayOutputStream and read (JDK)        avgt   10   1,343 ± 0,028  us/op
 6. InputStreamReader and StringBuilder (JDK)   avgt   10   6,980 ± 0,404  us/op
10. BufferedInputStream, ByteArrayOutputStream  avgt   10   7,437 ± 0,735  us/op
11. InputStream.read() and StringBuilder (JDK)  avgt   10   8,977 ± 0,328  us/op
 7. StringWriter and IOUtils.copy (Apache)      avgt   10  10,613 ± 0,599  us/op
 1. IOUtils.toString (Apache Utils)             avgt   10  10,605 ± 0,527  us/op
 3. Scanner (JDK)                               avgt   10  12,083 ± 0,293  us/op
 2. CharStreams (guava)                         avgt   10  12,999 ± 0,514  us/op
 4. Stream Api (Java 8)                         avgt   10  15,811 ± 0,605  us/op
 9. BufferedReader (JDK)                        avgt   10  16,038 ± 0,711  us/op
 5. parallel Stream Api (Java 8)                avgt   10  21,544 ± 0,583  us/op

```
Performance tests for big String (length = 50100), url in [github](https://github.com/Vedenin/useful-java-links/blob/master/helloworlds/5.0-other-examples/src/main/java/other_examples/ConvertBigStringToInputStreamBenchmark.java)[](https://i.stack.imgur.com/AYYhz.png) (mode = Average Time, system = Linux, score 200,715 is the best):

```
               Benchmark                        Mode  Cnt   Score        Error  Units
 8. ByteArrayOutputStream and read (JDK)        avgt   10   200,715 ±   18,103  us/op
 1. IOUtils.toString (Apache Utils)             avgt   10   300,019 ±    8,751  us/op
 6. InputStreamReader and StringBuilder (JDK)   avgt   10   347,616 ±  130,348  us/op
 7. StringWriter and IOUtils.copy (Apache)      avgt   10   352,791 ±  105,337  us/op
 2. CharStreams (guava)                         avgt   10   420,137 ±   59,877  us/op
 9. BufferedReader (JDK)                        avgt   10   632,028 ±   17,002  us/op
 5. parallel Stream Api (Java 8)                avgt   10   662,999 ±   46,199  us/op
 4. Stream Api (Java 8)                         avgt   10   701,269 ±   82,296  us/op
10. BufferedInputStream, ByteArrayOutputStream  avgt   10   740,837 ±    5,613  us/op
 3. Scanner (JDK)                               avgt   10   751,417 ±   62,026  us/op
11. InputStream.read() and StringBuilder (JDK)  avgt   10  2919,350 ± 1101,942  us/op
```
Graphs (performance tests depending on Input Stream length in Windows 7 system)
![image](https://user-images.githubusercontent.com/51409174/152540793-8ed04957-f735-43c7-9311-fc1ff9e1aabe.png)

Performance test (Average Time) depending on Input Stream length in Windows 7 system:

```
 length  182    546     1092    3276    9828    29484   58968

 test8  0.38    0.938   1.868   4.448   13.412  36.459  72.708
 test4  2.362   3.609   5.573   12.769  40.74   81.415  159.864
 test5  3.881   5.075   6.904   14.123  50.258  129.937 166.162
 test9  2.237   3.493   5.422   11.977  45.98   89.336  177.39
 test6  1.261   2.12    4.38    10.698  31.821  86.106  186.636
 test7  1.601   2.391   3.646   8.367   38.196  110.221 211.016
 test1  1.529   2.381   3.527   8.411   40.551  105.16  212.573
 test3  3.035   3.934   8.606   20.858  61.571  118.744 235.428
 test2  3.136   6.238   10.508  33.48   43.532  118.044 239.481
 test10 1.593   4.736   7.527   20.557  59.856  162.907 323.147
 test11 3.913   11.506  23.26   68.644  207.591 600.444 1211.545
```

Reply:
Pepsi challenge shows that execution is about the same. The GC really makes it tricky to profile memory, but StringUtil does appear to use less memory. I'll make the change. 
![image](https://user-images.githubusercontent.com/72526279/152554728-2d800075-206c-4b2b-86dc-3a9647a60027.png)

Reply:
Is pepsi challenge coming back as a common saying? I've seen it a few places recently.

Reply:
It never went away for me...